packages/db/src/lib/granule.ts

```diff
@@ -45,3 +48,49 @@ export const upsertGranuleWithExecutionJoinRecord = async (
   }
   return [granuleCumulusId];
 };
+
+/**
+ * Get cumulus IDs for all executions associated to a set of granules
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {Array<Object>} granules - array of granules with collectionId and granuleId
+ * @param {Object} [collectionPgModel] - Collection PG model class instance
+ * @param {Object} [granulePgModel] - Granule PG model class instance
+ * @param {Object} [granulesExecutionsPgModel]
+ *   Granules/executions PG model class instance
+ * @returns {Promise<number[]>}
+ */
+export const getApiGranuleExecutionCumulusIds = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<{ collectionId: string, granuleId: string }>,
+  collectionPgModel = new CollectionPgModel(),
+  granulePgModel = new GranulePgModel(),
+  granulesExecutionsPgModel = new GranulesExecutionsPgModel()
+): Promise<Array<number>> => {
+  const collectionMap: {[key: string]: number} = {};
+
+  const granuleCumulusIds: Array<number> = await Promise.all(granules.map(async (granule) => {
+    const { collectionId } = granule;
+    let collectionCumulusId = collectionMap[collectionId];
+    const { name, version } = deconstructCollectionId(granule.collectionId);
```

Comment:
Nit: I think this could could move inside the `if` because it's only necessary if we need to do the PG lookup. But not critical